api/v1alpha2/bundledeployment_types.go

```diff
@@ -69,8 +69,41 @@ type BundleDeploymentSpec struct {
 	//
 	// config is provisioner specific configurations
 	Config runtime.RawExtension `json:"config,omitempty"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be enabled or disabled when attempting to install the BundleDeployment.
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
```

Comment:
Do we need to add any of the kubebuilder validation tags to mark this field as optional as well?

Reply:
Yes, good catch! Updated in [commit](https://github.com/operator-framework/rukpak/pull/902/commits/3c22f6db5cca72ff64366a7928dcc3b10ca66dc6)